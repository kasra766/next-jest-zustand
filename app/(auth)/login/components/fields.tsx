"use client";
import { Button, Paper, Typography } from "@mui/material";
import { useForm } from "react-hook-form";
import { FormField } from "@/components/text-field";
import { useAuth } from "@/store/useAuth";

export function LoginFields() {
  const { login, signup } = useAuth();
  const { control, handleSubmit } = useForm({
    defaultValues: { user_name: "", password: "" },
  });

  const submit = (data: { user_name: string; password: string }) => {
    signup(data);
    login();
  };
  return (
    <Paper
      component="form"
      onSubmit={handleSubmit(submit)}
      elevation={12}
      className="flex min-w-[500px] max-w-[600px] flex-col gap-3 p-8"
    >
      <Typography variant="h4" mb={2} role="header">
        Login
      </Typography>

      <FormField
        control={control}
        name="user_name"
        rules={{ required: true }}
        type="text"
        autoFocus
        label="User Name"
        fullWidth
      />
      <FormField
        control={control}
        name="password"
        rules={{
          required: true,
          minLength: {
            value: 8,
            message: "password must be at least 8 character",
          },
        }}
        type="password"
        label="Password"
        fullWidth
      />
      <Button variant="contained" type="submit">
        submit
      </Button>
    </Paper>
  );
}
