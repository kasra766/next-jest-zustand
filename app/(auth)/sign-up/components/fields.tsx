"use client";

import { useForm } from "react-hook-form";
import { Button, Paper, Stack, Typography } from "@mui/material";
import { FormField } from "@/components/text-field";
import { type UserSignup, useAuth } from "@/store/useAuth";
import { emailPattern, mobilePattern, strongRegex } from "@/lib/patterns";

export function SignUpFields() {
  const { userSignup: defaultValues, login, signup } = useAuth();
  const { control, handleSubmit } = useForm<UserSignup>({ defaultValues });

  const submit = (data: UserSignup) => {
    signup(data);
    login();
  };

  return (
    <Paper
      component="form"
      onSubmit={handleSubmit(submit)}
      elevation={12}
      className="flex min-w-[500px] max-w-[600px] flex-col  gap-3 p-8"
    >
      <Typography variant="h4" mb={2}>
        Sign Up
      </Typography>
      <Stack direction={"row"} columnGap={2}>
        <FormField
          control={control}
          name="fname"
          rules={{
            required: { value: true, message: "name is required" },
          }}
          type="text"
          label="Name"
          autoFocus
        />
        <FormField
          control={control}
          name="lname"
          rules={{
            required: { value: true, message: "last name is required" },
          }}
          type="text"
          label="Last Name"
        />
      </Stack>
      <Stack direction={"row"} columnGap={2}>
        <FormField
          control={control}
          name="phone"
          rules={{
            required: { value: true, message: "phone is required" },
            pattern: {
              value: mobilePattern,
              message: "mobiles format is wrong",
            },
          }}
          type="tel"
          label="Phone"
        />
        <FormField
          control={control}
          name="user_name"
          rules={{
            required: { value: true, message: "user name is required" },
          }}
          type="text"
          label="User Name"
        />
      </Stack>
      <FormField
        control={control}
        name="email"
        rules={{
          required: { value: true, message: "email is required" },
          pattern: {
            value: emailPattern,
            message: "emails format is wrong",
          },
        }}
        type="email"
        label="Email"
      />

      <FormField
        control={control}
        name="password"
        rules={{
          required: { value: true, message: "password is required" },
          minLength: { value: 8, message: "min length muse be 8 character" },
          pattern: {
            value: strongRegex,
            message:
              "password must consist at least an Uppercase and lowercase and special character(@,$,...) and numbers",
          },
        }}
        type="password"
        label="Password"
      />

      <Button fullWidth type="submit" variant="contained">
        Submit
      </Button>
    </Paper>
  );
}
