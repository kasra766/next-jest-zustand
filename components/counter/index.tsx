"use client";
import { useCounter, useCounterStore } from "@/store/useCounter";
import { ButtonGroup, Button, Stack } from "@mui/material";

import AddIcon from "@mui/icons-material/Add";
import RemoveIcon from "@mui/icons-material/Remove";

export function Counter() {
  const { count, dec, inc } = useCounter();
  const increments = () => {
    inc();
  };
  return (
    <Stack justifyContent="center" alignItems={"center"} gap={3}>
      <p className="text-center" role="count_value">
        {count}
      </p>
      <ButtonGroup variant="contained">
        <Button
          onClick={inc}
          startIcon={<AddIcon />}
          variant="contained"
          data-testid="inc"
        >
          inc
        </Button>
        <Button
          onClick={dec}
          startIcon={<RemoveIcon />}
          variant="outlined"
          data-testid="dec"
        >
          dec
        </Button>
      </ButtonGroup>
    </Stack>
  );
}
