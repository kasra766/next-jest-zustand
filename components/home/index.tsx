"use client";
import { useAuth } from "@/store/useAuth";
import { Button, Paper, Stack, Typography } from "@mui/material";

export function Home() {
  const {
    user_information: { user_name },
    logout,
  } = useAuth();

  return (
    <Stack alignItems={"center"} pt={5}>
      <Paper
        elevation={12}
        className="flex min-w-[500px] max-w-[600px] flex-col gap-3 p-5"
      >
        <Typography
          variant="body1"
          textAlign={"center"}
          role="contentinfo"
        >{`welcome dear ${user_name}`}</Typography>
        <Button onClick={logout} variant="contained" role="button">
          Logout
        </Button>
      </Paper>
    </Stack>
  );
}
