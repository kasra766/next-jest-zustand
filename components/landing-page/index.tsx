"use client";
import Link from "next/link";
import { redirect } from "next/navigation";
import { Button, ButtonGroup, Stack, Typography } from "@mui/material";
import { useAuth } from "@/store/useAuth";

export function LandingPage() {
  const { isLogin } = useAuth();
  console.log(isLogin);

  if (isLogin) {
    redirect("/home");
  }
  return (
    <Stack justifyContent={"center"} alignItems={"center"} rowGap={3}>
      <Typography variant="h1">Welcome to over page</Typography>
      <Typography variant="body1">
        for seeing your profile please login or sign-up
      </Typography>
      <ButtonGroup size="large">
        <Button component={Link} href="/login" variant="contained">
          Login
        </Button>
        <Button component={Link} href="/sign-up" variant="outlined">
          Sign up
        </Button>
      </ButtonGroup>
    </Stack>
  );
}
