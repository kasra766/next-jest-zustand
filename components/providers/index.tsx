"use client";
import { SnackbarProvider } from "notistack";
import ThemeRegistry from "../theme-provider";

export function Providers({ children }: { children: React.ReactNode }) {
  return (
    <ThemeRegistry>
      <SnackbarProvider maxSnack={3}>{children}</SnackbarProvider>
    </ThemeRegistry>
  );
}
