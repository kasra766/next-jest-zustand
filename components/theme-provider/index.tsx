"use client";

import * as React from "react";
import { ThemeProvider } from "@mui/material/styles";
import CssBaseline from "@mui/material/CssBaseline";
import rtlPlugin from "stylis-plugin-rtl";
import NextAppDirEmotionCacheProvider from "./EmotionCache";
import theme from "./theme";

export default function ThemeRegistry({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <NextAppDirEmotionCacheProvider
      options={{ key: "mui", stylisPlugins: [rtlPlugin] }}
    >
      <ThemeProvider theme={theme}>
        {/* CssBaseline kickstart an elegant, consistent, and simple baseline to build upon. */}
        <CssBaseline />
        {children}
      </ThemeProvider>
    </NextAppDirEmotionCacheProvider>
  );
}
