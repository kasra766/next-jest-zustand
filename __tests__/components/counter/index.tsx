import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { Counter } from "@/components/counter";

describe("testing counter", () => {
  it("test counts value", () => {
    render(<Counter />);
    const count = screen.getByRole("count_value");
    expect(count).toHaveTextContent("0");
    const incBtn = screen.getByTestId("inc");
    fireEvent.click(incBtn);
    expect(count).toHaveTextContent("1");

    const decBtn = screen.getByTestId("dec");
    fireEvent.click(decBtn);
    expect(count).toHaveTextContent("0");
  });
});

describe("snap shot of counter", () => {
  it("snap shot", () => {
    const component = renderer.create(<Counter />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
