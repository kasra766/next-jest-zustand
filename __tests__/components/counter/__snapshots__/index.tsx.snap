// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snap shot of counter snap shot 1`] = `
<div
  className="MuiStack-root css-nm050a-MuiStack-root"
>
  <p
    className="text-center"
    role="count_value"
  >
    0
  </p>
  <div
    className="MuiButtonGroup-root MuiButtonGroup-contained css-zqcytf-MuiButtonGroup-root"
    role="group"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
      data-testid="inc"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      inc
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      data-testid="dec"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RemoveIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13H5v-2h14v2z"
          />
        </svg>
      </span>
      dec
    </button>
  </div>
</div>
`;
